src/kdtree.rs

```diff
@@ -288,6 +288,25 @@ impl<A: Float + Zero + One, T, U: AsRef<[A]>> KdTree<A, T, U> {
         }
     }
 
+    pub fn remove(&mut self, point: &U, data: &T) -> Result<(), ErrorKind> {
+        if let Err(err) = self.check_point(point.as_ref()) {
+            return Err(err);
+        }
+        if let (Some(points), Some(bucket)) = (self.points.take(), self.bucket.take()) {
+            self.points = Some(points.into_iter().filter(|x| x != point).collect());
```

Comment:
trying to understand the safety of this implementation, what if point matches node A but data matches node B?

Reply:
If the incorrect point/data pair are passed in, there aren't any checks that they match. I didn't think there was a way to validate that. If they indices match we could use that, but I didn't think that was always true.

Reply:
related to the use case comment - iirc the points and buckets match by index; then we should be able to iterate by index to ensure both `point` and `data` match the input